coriolis/osmorphing/redhat.py

```diff
@@ -105,12 +127,27 @@ def _get_ifcfgs_by_type(self, ifcfg_type):
                 ifcfgs.append((ifcfg_file, ifcfg))
         return ifcfgs
 
-    def set_net_config(self, nics_info, dhcp):
-        ifcfgs_ethernet = self._get_ifcfgs_by_type("Ethernet")
+    def _write_nic_configs(self, nics_info):
+        for idx,_ in enumerate(nics_info):
+            dev_name = "eth%d" % idx
+            cfg_path = "etc/sysconfig/network-scripts/ifcfg-%s" % dev_name
+            if self._test_path(cfg_path):
+                self._exec_cmd_chroot(
+                    "cp %s %s.bak" % (cfg_path, cfg_path)
+                )
+            self._write_file_sudo(
+                cfg_path,
+                IFCFG_TEMPLATE % {
```

Comment:
Couldn't we also add the `HWADDR` (MAC address) option here (granted we know it and the dest provider supports setting it on NICs ofc)

Reply:
the point of this change is to create a valid config when we don't know the MAC or are unable to retain the NIC name because we cannot retain both MAC and IP. The ordering of eth0,1,2,3 is not guaranteed across reboots, hence, the flat, generic configs that should work across reboots even if eth0 decides to become eth1 after a reboot.

Also, I think that if we want to create a generic approach to something, we need to leave out things that are particular to one component or another, and be as generic as possible without affecting functionality. In this instance, all NICs are set to DHCP, regardless of their MAC.